portfolio/src/main/java/com/google/sps/servlets/FunnyServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+/** Handles request sent to /funny */
+@WebServlet("/funny")
+public class FunnyServlet extends HttpServlet {
+  private ArrayList<String> jokeArray = new ArrayList<String>();
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if(jokeArray.isEmpty()) {
```

Comment:
So, this is in memory on the servlet. It doesn't really persist between requests so you'll just be creating this array on every request (which is fine) and I don't think that's what you were trying to do.

You could just initialize a private static final List<String> and use that. I don't recall the syntax to do this off the top of my head but I'll find it for you.

It's all fine as-is, just wanted to clarify that state won't stick around for subsequent requests.

Reply:
That's a good point, I think that's better actually.